txnkv/txnsnapshot/snapshot.go

```diff
@@ -203,6 +205,13 @@ func (s *KVSnapshot) BatchGet(ctx context.Context, keys [][]byte) (map[string][]
 	ctx = context.WithValue(ctx, retry.TxnStartKey, s.version)
 	bo := retry.NewBackofferWithVars(ctx, batchGetMaxBackoff, s.vars)
 
+	if s.interceptor != nil {
```

Comment:
Seems that there are some places missed calling `interceptor`, for example, batch coprocessor. Although currently we don't need it, it would be great to have that from the perspective of client-go. If don't know how to deal with it correctly, then at least add a TODO.

Reply:
I think users need to know that they are using `KVTxn` or `KVSnapshot`, and the effective area of `Interceptor` is limited to a specific `KVTxn` or `KVSnapshot`.
So when users don't need to use abstract transactions or snapshots, but need to send requests directly (e.g. copr), they need to **explicitly** add the `Interceptor` to the `ctx` associated with the request. That's why we need to provide `CtxWithInterceptor` API to users.

Reply:
In the implementation of SQL Exec Count, we have done this to count the distribution of copr requests.

Reply:
Do you mean that actually batch coprocessor is not handled by client-go? Then I think it would be fine.

Reply:
All coprocessor requests will eventually be processed by `ClientHelper` in client-go, but `ClientHelper` is just a simple wrap of inner client, and only provide a `SendReq` method. In this pattern, I think it is reasonable to let user explicitly call `WithCtxInterceptor` on `ctx`.

The `ctx` of transaction and snapshot is constructed internally, so we provide high-level `AddInterceptor()` for transaction and snapshot; The `ctx` requested by the coprocessor is constructed by the user and passed in through `ClientHelper.SendReq(ctx)`, so we let user to call `WithCtxInterceptor` explicitly.

There is a unified principle here: Where the `context.Context` constructed, where the `Interceptor` bound.